File: notification-service/src/main/java/dev/zvaryyka/notificationservice/repository/NotificationStatusRepository.java
package dev.zvaryyka.notificationservice.repository;

import dev.zvaryyka.notificationservice.models.NotificationStatus;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface NotificationStatusRepository extends JpaRepository<NotificationStatus,Integer> {

    NotificationStatus findByStatusName(String statusName);
}

File: rebalancer-service/src/main/kotlin/dev/zvaryyka/rebalancerservice/repository/NotificationStatusRepository.kt
package dev.zvaryyka.rebalancerservice.repository

import dev.zvaryyka.rebalancerservice.models.NotificationStatus
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

@Repository
interface NotificationStatusRepository : JpaRepository<NotificationStatus, Int> {
    fun findByStatusName(statusName: String) : NotificationStatus
}